src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -186,7 +199,7 @@ create_distributed_table(PG_FUNCTION_ARGS)
 
 		ConvertToDistributedTable(relationId, distributionColumnName,
 								  distributionMethod, REPLICATION_MODEL_COORDINATOR,
-								  INVALID_COLOCATION_ID);
+								  INVALID_COLOCATION_ID, allowEmpty);
```

Comment:
(i.e. `allowEmpty` is only useful when false, to mean _empty is not allowed_. But the semantic we wish to denote is _non-empty is not allowed_, otherwise phrased as `not("empty is required")`. So `requireEmpty` seems better.